src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -921,18 +942,71 @@ export class ConnectionDialogWebviewController extends ReactWebviewPanelControll
                             this.state.connectionStatus = ApiStatus.Error;
                             this.state.trustServerCertError =
                                 result.errorMessage;
+
+                            // connection failing because the user didn't trust the server cert is not an error worth logging;
+                            // just prompt the user to trust the cert
+
                             return state;
                         }
+
                         this.state.formError = result.errorMessage;
                         this.state.connectionStatus = ApiStatus.Error;
+
+                        sendActionEvent(
+                            TelemetryViews.ConnectionDialog,
+                            TelemetryActions.CreateConnection,
+                            {
+                                result: "connectionError",
+                                errorNumber: String(result.errorNumber),
+                                newOrEditedConnection: this
+                                    ._connectionToEditCopy
+                                    ? "edited"
+                                    : "new",
+                                connectionInputType:
+                                    this.state.selectedInputMode,
+                                authMode:
+                                    this.state.connectionProfile
+                                        .authenticationType,
+                            },
+                        );
+
                         return state;
                     }
                 } catch (error) {
                     this.state.formError = getErrorMessage(error);
                     this.state.connectionStatus = ApiStatus.Error;
+
+                    sendErrorEvent(
+                        TelemetryViews.ConnectionDialog,
+                        TelemetryActions.CreateConnection,
+                        error,
```

Comment:
PIIs?

Reply:
`includeErrorMessage` defaults to `false`, so I think this'll be okay, but I'll explicitly set that to be sure